src/node/domains/mod.rs

```diff
@@ -0,0 +1,201 @@
+    domain_service::FullClient<system_domain_runtime::RuntimeApi, ExecutorDispatch>;
+pub(crate) type NewFull = domain_service::NewFull<
+    Arc<FullClient>,
+    sc_executor::NativeElseWasmExecutor<ExecutorDispatch>,
+    subspace_runtime_primitives::opaque::Block,
+    crate::node::FullClient,
+    system_domain_runtime::RuntimeApi,
+    ExecutorDispatch,
+>;
+/// Chain spec of the secondary chain
+pub type ChainSpec = sc_subspace_chain_specs::ExecutionChainSpec<ExecutionGenesisConfig>;
+
+/// Secondary executor node
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
+pub struct SecondaryNode {
+    #[derivative(Debug = "ignore")]
+    client: Weak<FullClient>,
+    core: Option<CoreNode>,
+    _rpc_handlers: crate::utils::Rpc,
+}
+
+impl SecondaryNode {
+    pub(crate) async fn new(
+        cfg: Config,
+        directory: impl AsRef<Path>,
+        chain_spec: ChainSpec,
+        primary_new_full: &mut crate::node::NewFull,
+    ) -> anyhow::Result<Self> {
+        let Config { base, relayer_id: maybe_relayer_id, core } = cfg;
+        let service_config =
+            base.configuration(directory.as_ref().join("system"), chain_spec).await;
+
+        let secondary_chain_config = DomainConfiguration { service_config, maybe_relayer_id };
+        let imported_block_notification_stream = primary_new_full
+            .imported_block_notification_stream
+            .subscribe()
+            .then(|imported_block_notification| async move {
+                (
+                    imported_block_notification.block_number,
+                    imported_block_notification.fork_choice,
+                    imported_block_notification.block_import_acknowledgement_sender,
+                )
+            });
+
+        let new_slot_notification_stream = primary_new_full
+            .new_slot_notification_stream
+            .subscribe()
+            .then(|slot_notification| async move {
+                (
+                    slot_notification.new_slot_info.slot,
+                    slot_notification.new_slot_info.global_challenge,
+                )
+            });
+
+        let (gossip_msg_sink, gossip_msg_stream) =
+            sc_utils::mpsc::tracing_unbounded("Cross domain gossip messages");
+
+        // TODO: proper value
+        let block_import_throttling_buffer_size = 10;
+
+        let secondary_chain_node = domain_service::new_full(
+            secondary_chain_config,
+            primary_new_full.client.clone(),
+            primary_new_full.network.clone(),
+            &primary_new_full.select_chain,
+            imported_block_notification_stream,
+            new_slot_notification_stream,
+            block_import_throttling_buffer_size,
+            gossip_msg_sink.clone(),
+        )
+        .await?;
+
+        let mut domain_tx_pool_sinks = std::collections::BTreeMap::new();
+
+        let core = if let Some(core) = core {
+            CoreNode::new(
+                core,
+                directory.as_ref().join("core"),
```

Comment:
I suggest making the directory specific to each core domain id `directory.as_ref().join(format!("core-{}", core_domain_id))` in case you run another core domain after one core domain is started under the same directory.